src/cdi/CPCommon.hh

```diff
@@ -0,0 +1,48 @@
+//----------------------------------*-C++-*----------------------------------//
+/*!
+ * \file   cdi/CPCommon.hh
+ * \author Kendra Long
+ * \date   Mon Jan 19 13:41:01 2001
+ * \brief  Datatypes needed in CPEloss classes
+ * \note   Copyright (C) 2016-2019 Los Alamos National Securty, LLC.
+ *         All rights reserved. */
+//---------------------------------------------------------------------------//
+
+#ifndef __cdi_CPCommon_hh__
+#define __cdi_CPCommon_hh__
+
+#include <cstdint>
+
+namespace rtt_cdi {
+
+namespace constants {
+//! Number of models contained in rtt_cdi::CPModel.
+unsigned int const num_CPModels(3);
+} // namespace constants
+
+enum class CPModel {
+  ELOSS = 0,        /*!< energy loss model */
+  LA_SCATTERING = 1 /*!< large-angle scattering model */
+};
+
+enum class CPModelType {
+  UNASSIGNED_ETYPE =
+      0, /*!< unassigned type; used as a placeholder before deciding type */
+  ANALYTIC_ETYPE = 1, /*!< an Analytic eloss model */
+  TABULAR_ETYPE = 2   /*!< an Analytic eloss model */
+};
+
+// Basic data required to describe a charged particle
+class CParticle {
+private:
+  int32_t zaid;
+  double mass;
+
+public:
+  int32_t get_zaid() const { return zaid; }
+  double get_mass() const { return mass; }
+};
```

Comment:
Just a thought: what about adding a vector of references to CPEloss classes (and later classes containing large angle scattering physics) here, so that with this class you get both the charged particle properties as well as the reactions it can undergo? Might make bookkeeping more convenient later.

Reply:
Hmm, this is an interesting idea! I guess in my original plan, the CParticle class was intended to provide the necessary input to build a CDI Eloss -- I hadn't thought about using it (or something like it) as a container. My general plan was to implement the Eloss container part of the equation in the calling code, but I'm certainly not tied to that idea if you think it makes more sense

(Also, it just occurred to me that I haven't even provided a way to set the zaid/mass, so obviously this design isn't terribly useful as is. I guess this is what happens when the only actual Eloss model you've implemented needs no particle parameters.)

Reply:
My thinking in bringing this up was just to avoid lookups-by-ZAID during compute loops; initializing a CParticle for each physical particle, with references to physics classes ready to go, that is then referenced would be one way around that I think. But there's a bit of a circular dependency issue here, because you kind of need CParticle to initialize CDI Eloss, like you say... unfortunately circular dependencies that don't compile is my natural C++ coding style. Maybe a separate CParticlePhysics class that holds references to both a CParticle and CDI Elosses would be best? Whatever you prefer layout-wise is fine with me, I just think it would be nice to only find and store one reference by zaid when creating e.g. a computational particle (you may have had this in mind all along).

Reply:
I agree, I think that's the way to go for the actual transport step--  I'm just not sure where the container of particle info / eloss references should live. Right now I'm assuming this container is "above" draco in the calling code, but I'm open to other design options :)

Reply:
Oh good point, that makes way more sense, leaving those details out of Draco -- please ignore what I said in this thread.

Reply:
Heh, no worries. In any case, the CParticle class will definitely need to change so you can... you know... actually set the class members :D